src/main/java/org/mule/extension/runtime/ExpirationPolicy.java

```diff
@@ -0,0 +1,35 @@
+package org.mule.extension.runtime;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A policy around when should a given instances be considered elegible for expiration.
+ * <p/>
+ * Notice that this contract is not directly tied to the instance to be expired itself. It's tied
+ * to the concept of idle based expiration.
+ *
+ * @since 1.0
+ */
+public interface ExpirationPolicy
+{
+
+    /**
+     * Determines if an instance should be expired based on the last time it was used.
+     *
+     * @param lastUsed a scalar time value
+     * @param timeUnit a {@link TimeUnit} that qualifies the {@code lastUsed}
+     * @return {@code true} if expiration should take place. {@code false} otherwise.
+     */
+    boolean isExpired(long lastUsed, TimeUnit timeUnit);
```

Comment:
i like it like this. I think it's easier to use and read, it also saves you exercise of go checking the javadoc to see in which unit the time is.... I like it :)